ngraph/core/src/preprocess/pre_post_process.cpp

```diff
@@ -260,73 +284,202 @@ OutputInfo::OutputInfo(OutputInfo&&) noexcept = default;
+            m_inputs[0] = std::move(builder);
+        }
+    }
+
+    // TODO: this is created for compatibility, consider to remove
+    void add_output_info(OutputInfo&& builder) {
+        if (builder.m_impl->m_has_index) {
+            OPENVINO_ASSERT(builder.m_impl->m_index < m_inputs.size(), "Output index is out of range");
+            m_outputs[builder.m_impl->m_index] = std::move(builder);
+        } else if (builder.m_impl->m_has_name) {
+            size_t index = find_output_index(builder.m_impl->m_name);
+            m_outputs[index] = std::move(builder);
+        } else {
+            OPENVINO_ASSERT(1 == m_inputs.size(), "Function shall have only one output");
+            m_outputs[0] = std::move(builder);
+        }
+    }
+    std::vector<InputInfo> m_inputs;
+    std::vector<OutputInfo> m_outputs;
+    std::shared_ptr<Function> m_function = nullptr;
 };
 
-PrePostProcessor::PrePostProcessor() : m_impl(std::unique_ptr<PrePostProcessorImpl>(new PrePostProcessorImpl())) {}
+PrePostProcessor::PrePostProcessor(const std::shared_ptr<Function>& function)
+    : m_impl(std::unique_ptr<PrePostProcessorImpl>(new PrePostProcessorImpl(function))) {}
 PrePostProcessor::PrePostProcessor(PrePostProcessor&&) noexcept = default;
 PrePostProcessor& PrePostProcessor::operator=(PrePostProcessor&&) noexcept = default;
 PrePostProcessor::~PrePostProcessor() = default;
 
+InputInfo& PrePostProcessor::input() {
+    OPENVINO_ASSERT(m_impl->m_inputs.size() == 1,
+                    "PrePostProcessor::input() - function must have exactly one input, got ",
+                    m_impl->m_inputs.size());
+    return m_impl->m_inputs.front();
+}
+
+InputInfo& PrePostProcessor::input(size_t input_index) {
+    OPENVINO_ASSERT(m_impl->m_inputs.size() > input_index,
+                    "PrePostProcessor::input(size_t) - function doesn't have input with index ",
+                    input_index,
+                    ". Total number of inputs is ",
+                    m_impl->m_inputs.size());
+    return m_impl->m_inputs[input_index];
+}
+
+InputInfo& PrePostProcessor::input(const std::string& tensor_name) {
+    size_t index = m_impl->find_input_index(tensor_name);
+    return m_impl->m_inputs[index];
+}
+
+OutputInfo& PrePostProcessor::output() {
+    OPENVINO_ASSERT(m_impl->m_outputs.size() == 1,
+                    "PrePostProcessor::output() - function must have exactly one output, got ",
+                    m_impl->m_outputs.size());
+    return m_impl->m_outputs.front();
+}
+
+OutputInfo& PrePostProcessor::output(size_t output_index) {
+    OPENVINO_ASSERT(m_impl->m_outputs.size() > output_index,
+                    "PrePostProcessor::output(size_t) - function doesn't have input with index ",
+                    output_index,
+                    ". Total number of inputs is ",
+                    m_impl->m_inputs.size());
+    return m_impl->m_outputs[output_index];
+}
+
+OutputInfo& PrePostProcessor::output(const std::string& tensor_name) {
```

Comment:
do we have tests that getting by tensor name and set precision, then get the same input by index and set layout  - sets both layout and precision?